com.unity.sg2/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs

```diff
@@ -39,6 +39,67 @@ public override IReadOnlyList<ItemLibraryDatabaseBase> GetVariableTypesDatabases
             return databases;
         }
 
+        public override IReadOnlyList<ItemLibraryDatabaseBase> GetGraphElementContainerDatabases(GraphModel graphModel, IGraphElementContainer container)
+        {
+            if (container is not GraphDataContextNodeModel contextNode || graphModel is not ShaderGraphModel sgModel)
+            {
+                return base.GetGraphElementContainerDatabases(graphModel, container);
+            }
+
+            var availableBlocks = new List<ItemLibraryItem>();
+            var handler = sgModel.GraphHandler.GetNode(contextNode.graphDataName);
+
+            foreach (var portHandler in handler.GetPorts())
+            {
+                if (!portHandler.IsInput || !portHandler.IsHorizontal) continue;
```

Comment:
we probably want to remove the CLDS port/node handler references here as this is UI code